website/docs/opengl/c1/2-hello-quad.md

```diff
@@ -0,0 +1,524 @@
+Here is the part where we actually give the GPU some data it can work with. For 95% of graphics applications, a vertex buffer is **required** to display anything on screen.
+
+### What is a vertex buffer?
+Before we can continue, it's handy to know what a vertex buffer is, and what it does.
+
+Let's define both vertex, and buffer.
+
+* Vertex - A point where two lines, or edges meet.
+* Buffer - A region in memory that can be accessed to be written to, or read from.
+
+Knowing these definitions should give us a good idea of what a vertex buffer is. It's a region of memory, on the GPU, that stores points, which will get **rasterized** on-screen (usually in the form of triangles).
+
+### Vertex data
+Let's create the data that we will fill the vertex buffer with!
+
+In your `OnLoad` function, just after you call `_gl.BindVertexArray()`, add the following:
+
+```cs
+float[] vertices =
+{
+     0.5f,  0.5f, 0.0f,
+     0.5f, -0.5f, 0.0f,
+    -0.5f, -0.5f, 0.0f,
+    -0.5f,  0.5f, 0.0f
+};
+```
+
+### What makes up a quad?
+In modern graphics programming, you are expected to use triangles. While you may see options for quads, or lines, these are mostly deprecated.
+
+Therefore, a quad is made of two right-angle triangles. This can best be seen if we view the result in **wireframe** mode.
+
+![Wireframe quad](../../../images/opengl/chapter1/wireframe-quad.png)
+
+In the image, you can also see where the four vertices go in relation to the quad. While you won't *usually* be defining vertices yourself, it's still handy to know how it works.
+
+### Creating the buffer
+Now that we've got our vertex data, let's create the buffer!
+
+At the top of your class, add the following:
+
+```cs
+private static uint _vbo;
+```
+
+Then, in your `OnLoad` method, under where the vertices are defined, add:
+
+```
+_vbo = _gl.GenBuffer();
+_gl.BindBuffer(BufferTargetARB.ArrayBuffer, _vbo);
+```
+
+Much like the VAO, we need to first generate the buffer, and then bind it. Unlike vertex arrays, however, buffers need to be bound to a **target**. This allows you to bind buffers to different targets at the same time. Some of the more common targets include:
+
+* `ArrayBuffer` - The vertex buffer target (which is what we're using here).
+* `ElementArrayBuffer` - The element array buffer target (which you'll see later).
+* `UniformBuffer` - A uniform buffer, not used in this tutorial, but we'll get to it in a later tutorial.
+
+In this case, we're binding to `ArrayBuffer` since we're creating a vertex buffer.
+
+### Filling our buffer with data
+Let's fill our buffer with some data! Before we do that though, you need to be aware of `unsafe` in C#.
+
+#### Unsafe C#
+Silk.NET heavily uses `unsafe` code. Don't worry, this won't make your computer explode, however it does exit out of the "safe" managed environment of C#, and enters a realm where undefined behaviour, segmentation faults, and strange results are more likely to occur if you are not careful.
```

Comment:
```suggestion
Silk.NET heavily uses `unsafe` code. Don't worry, this won't make your computer explode, however it does exit out of the "memory safe" managed environment of C#, and enters a realm where undefined behavior, segmentation faults, and strange results are more likely to occur if you are not careful. Since we're working with low-level APIs, and OpenGL is defined in plain C, some unsafe code will be necessary in order for us to be able to communicate with it from C#.
```

Reply:
are we sure we want american english here? there's [precedent](https://github.com/dotnet/Silk.NET/blob/93f987aa40f99c3b6e347d3863fe2e524c2a9398/examples/CSharp/OpenGL%20Tutorials/Tutorial%203.2%20-%20Diffuse%20Lighting/Program.cs#L122) for both british and american english

Reply:
I'm just gonna link [this commit](https://github.com/dotnet/Silk.NET/commit/4ae5a303e4238f3ed4fae199b6e64bb5ab2c0d0c)...

Reply:
ah, so is american english the standard in Silk? if so we should really do another British ->American english pass (maybe we should write something into the bakery to convert binded function names and enum names to american english :^)

Reply:
Wouldn't hurt to do so, but I don't think it's all that important either. We could always just ctrl-F "colour" through the entire project.

That commit still makes me chuckle lol

Reply:
CAN WE PLEASE USE BRITISH ENGLISH PLEASE IT MAY NOT BE NORMAL IN OUR CODE BUT IT IS DEFINITELY THE PRECEDENT IN OUR LITERATURE SEE ALL OF THE BLOG POSTS USING BRITISH ENGLISH SURE THAT'S BECAUSE I WROTE THEM BUT THE POINT STILL STANDS BRITISH ENGLISH PLEASE THANK YOU THE ONLY REASON WE DON'T USE IT IN CODE IS BECAUSE WE'RE FOLLOWING MICROSOFT'S PRECEDENT THERE KTHX BYE

Reply:
happy to change to british english if desired